app/static/js/main.js

```diff
@@ -276,29 +315,32 @@ function addPolygonToList(polygonID, computation) {
     $("#show-hide-" + polygonID).on("click", function(e) {
         var polygon = polygons.collection[polygonID];
         showHidePolygonButton(this, polygon);
+        isHidden = !isHidden;
     })
     $("#delete-" + polygonID).on("click", function(e) {
         var polygonID = $(this).parent().attr("id");
         var polygon = polygons.collection[polygonID];
         deletePolygonButton(this, polygon);
     })
     $("#" + polygonID).on("click", function(e) {
-       clearPolygonListBorders();
        var polygon = polygons.collection[polygonID];
-       polygons.setSelection(polygon);
-       createPolygonListBorder(polygonID);
+       if (!polygons.isInSCollec(polygon) && !isHidden) {
```

Comment:
Does this still work how that it has been refactored?
``` 
 polygons.isInSCollec()
```

Either way it should also be renamed.